odbc/src/api/odbc_uri.rs

```diff
@@ -0,0 +1,191 @@
+use crate::errors::{ODBCError, Result};
+use std::collections::BTreeMap;
+
+// TODO SQL-990: These errors will probably change.
+const NOT_EMPTY_ERROR: &str = "uri must not be empty";
+const EQUAL_ERROR: &str = "all uri atttributes must be of the form key=value";
+
+// TODO SQL-990: Audit these mandatory attributes
+const USER: &[&str] = &["user", "uid"];
+const PWD: &[&str] = &["pwd", "password"];
+const SERVER: &[&str] = &["server"];
+const SSL: &[&str] = &["ssl"];
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct ODBCUri<'a>(BTreeMap<String, &'a str>);
```

Comment:
sadly, the lowercasing of the keys means that the ODBCUri must own the keys :(

If we really care we could use ` Cow` here so that it only owns the memory when lowercasing actually occurs.